internal/services/balance.go

```diff
@@ -0,0 +1,93 @@
+package services
+
+import (
+	"context"
+	"sort"
+
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/database"
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/models"
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/utils"
+)
+
+type BalanceService struct {
+	storage balanceStorage
+}
+
+type balanceStorage interface {
+	FindAccrualFlow(ctx context.Context, userID string) (*[]database.AccrualFlowItemDB, error)
+
+	CreateWithdrawal(ctx context.Context, orderID, userID string, amount float64) error
+
+	FindWithdrawalFlow(ctx context.Context, userID string) (*[]database.WithdrawalFlowItemDB, error)
+}
+
+func NewBalanceService(storage balanceStorage) *BalanceService {
+	return &BalanceService{storage: storage}
+}
+
+func (b *BalanceService) GetUserBalance(ctx context.Context, userID string) (models.Balance, error) {
```

Comment:
Возможно стоит вернуть указатель на модель? Так будет проще интерпретировать отсутствие значения.
```suggestion
func (b *BalanceService) GetUserBalance(ctx context.Context, userID string) (*models.Balance, error) {
```